src/lib.rs

```diff
@@ -274,7 +273,15 @@ pub fn get_repository_versions_response(
     let mut body = Vec::new();
     decoder.read_to_end(&mut body)?;
 
-    let signed = Signed::decode(body.as_slice())?;
+    get_repository_versions_data(&body, public_key)
+}
+
+/// Parse a signed binary message containing all of the packages on the package registry.
+pub fn get_repository_versions_data(
```

Comment:
This doesn't call a get API endpoint, so let's instead call these something like `parse_repository_v2_versions`?

Might be good to link to the docs: https://github.com/hexpm/specifications/blob/main/registry-v2.md